usecases/schema/tenant.go

```diff
@@ -199,15 +199,16 @@ func (h *Handler) UpdateTenants(ctx context.Context, principal *models.Principal
 //
 // Class must exist and has partitioning enabled
 func (h *Handler) DeleteTenants(ctx context.Context, principal *models.Principal, class string, tenants []string) error {
-	if err := h.Authorizer.Authorize(principal, authorization.DELETE, authorization.Shards(class, tenants...)...); err != nil {
-		return err
-	}
 	for i, name := range tenants {
 		if name == "" {
 			return fmt.Errorf("empty tenant name at index %d", i)
 		}
 	}
 
+	if err := h.Authorizer.Authorize(principal, authorization.DELETE, authorization.Shards(class, tenants...)...); err != nil {
+		return err
+	}
+
```

Comment:
**Issue (blocking):** The `Authorize` step has to be the first thing that is called in the overall handler logic. (In an ideal world it would be a middleware but we're not there yet!)

By calling it after looping through the `tenants` we are leaking information, regardless of how trivial it may seem, to whoever is making the forbidden request. Can the logic of `authorization.Shards` not be changed to not modify the `[]string` slice passed to it?